engine/src/main/java/org/hibernate/search/engine/common/spi/SearchIntegration.java

```diff
@@ -21,11 +24,15 @@ public interface SearchIntegration extends AutoCloseable {
 
 	IndexManager indexManager(String indexManagerName);
 
+	SavedState saveForRestart();
```

Comment:
Actually we need both the saved state and the search integration (that generates it) in the `SearchIntegrationPartialBuildStateImpl`.
We could try something different to not expose this method on SPI, for instance we can have a `SearchIntegrationImpl` instead of `SearchIntegration` as field in `SearchIntegrationPartialBuildStateImpl`.
I'm following the latter solution.